pshtt.py

```diff
@@ -310,10 +323,11 @@ def https_check(endpoint):
             endpoint.https_expired_cert = True
 
         # Check for whether there's a valid chain to Mozilla.
+        # Note: this will also catch expired certs, but this is okay.
         if (
             (("Mozilla NSS CA Store") in msg) and
             (("FAILED") in msg) and
-            (("unable to get local issuer certificate") in msg)
+            (("Certificate is NOT Trusted") in msg)
```

Comment:
This was previously not detecting the use of a self-signed certificate. I've made the catching logic more general for chain errors, at the cost of also including expired certificates. So if a certificate is expired, we'll now flag both the `HTTPS Bad Chain` _and_ `HTTPS Expired Cert` fields as `False`. I'm fine with that, personally.